### PR TITLE
Loot Table Overhaul

### PR DESCRIPTION
**Pull request by JohnWildkins**
### **Loot Table Overhaul**
_or: How I Learned To Stop Worrying And Love The Hunter's Guild_

---

TL;DR: Drop rates across the board have increased. Treasure Hunter now only increases your chance of acquiring rare loot, it does not (usually) _decrease_ your chance of mundane loot. In general, most creatures had two chances to drop an item, at 40% chance per 'roll'. Almost universally this has been increased to 100% for the first roll, and 40% for the second. This means there are few (if any) creatures that **drop nothing** on death.

Specific changes:

### CZ220
- **Colicoid Experiment**
  - 1x95% => 1x100%
  - Credit drop chance 80% => 100%/60%
 - **Malfunctioning Patrol Droid**
   - 100% => 100%/40%
 - **Malfunctioning Probe Droid**
   - 2x60% => 100%/40%
 - **Mynock**
   - 2x75% => 100%/50%

### Viscara
  - **Mandalorian Ranger**
    - Sweet Butter no longer classed as 'rare'
  - **Mandalorian Hunter/Scout**
    - Rare chance to drop Mando Facility map
  - **Mandalorian Crates**
    - Unflagged electroblade (fmr. lightsaber) parts as rare, increased drop chance from 5/138 to 10/143
  - **Outlaw (Trandoshan)**
    - 2x40% => 100%/40%
  - **Outlaw (Human)**
    - 1x50% => 100%/50%
  - **Gimpassa Hatchling**
    - 2x60% => 100%/60%
    - Rare Table added @ 20%
      - Gimpassa Blood and Shells moved there as 2/3 and 1/3 chance respectively
  - **Kinrath**
    - 1x40% => 100%/40%
    - Rare Table: Silk (2/3), Limb (1/3), 20% drop chance
    - Silk removed from normal drop table
    - Normal Table limb freq. dropped from 5 to 1 (5/40 => 1/26)
  - **Cairnmog**
    - 2x40% => 100%/40%
    - Rare Table @ 20%
    - Blood, Teeth, Map moved to Rare Table at ratio 2:4:1
    - Normal Table chances set to 10:10:5:10 for meat, spine, T1 and T2 leather
  - **Cairnmog Alpha**
    - 2x60% => 100%/60%, Rares @ 30%
    - Changes as above
  - **Nashtah**
    - 2x40% => 100%/40%, Rare @ 20%
    - Moved Tail and Map to Rares @ 3:1 ratio
  - **Crystal Spider**
    - 2x30% => 100%
    - RDT 5% => 20%, Agate moved to Rare Table @ 3:1 ratio to map
  - **Kath Hound**
    - 2x75% => 100%/50%
  - **Warocas**
    - 2x75% => 100%/50%
    - Removed rare flag from normal table warocas leg
  - **Raivor**
    - 2x40% => 100%/40%
    - RDT @ 20%: Scale/Blood/Map @ 2:4:1
  - **Flesheater**
    - 2x40% => 100%/40%
    - Coxxion Metal moved to RDT
    - RDT 5% => 10%: 20:10:4:1 Coxxion Metal / Babonsch / Coxxion Base Map / Swamplands Map
  - **Fleshleader**
    - 1x95%, 2x20% => 1x100%, 2x20%
    - RDT 5% => 20%

### Korriban
Note: Provisional new names as of #1508, loot item names may change accordingly
  - **Tuk'ata (Pelko Bug Swarm)**
    - 2x40% => 100%/40%
    - Ruined Leather wt. 10 => 5
    - Flawed Leather wt. 2 => 10
  - **Hssiss (K'lor'slug)**
    - 2x40% => 100%/40%
    - Skin moved to RDT @ 10%
    - Ruined Leather wt. 25 => 5
  - **Shyrack**
    - 2x40% => 100%/40%
    - Shyrack Meat wt. 50/100 => 30/80
  - **Moraband Serpent**
    - 2x40% => 100%/40%
    - Ruined Leather wt. 10/35 => 5/35
    - Flawed Leather wt. 5/35 => 10/35
  - **Terentatek (Wraid)**
    - 2x40% => 100%/40%
    - Spine to RDT @ 5%; was 1/65
    - Ruined Leather removed
  - **Sith Ghost (Possessed Apprentice)**
    - 2x40% => 100%/40%
    - Ruined Leather wt. 20/109 => 5/71
    - Map, Abdamaryllia, and blueprints to RDT @ 10%, ratio 2:1:1 (split across 11); was 10/109, 2/109, and 1/109

### Mon Calamari

  - **Amphi-Hydrus**
    - 2x40% => 100%/40%
    - Reduced map weighting from 8/9 to 4/5 (agate 1/9 => 1/5)
  - **Aradile**
    - 2x40% => 100%/40%
    - Skin moved to RDT @ 5% (was 2/45)
  - **Viper**
    - 2x40% => 100%/40%
    - Flawed Leather wt. 10/40 => 5/35
  - **Eco-Terrorist**
    - 2x40% => 100%/40%
    - Armor drops moved to separate RDT from maps, @ 10%/0% (treasure hunter bait)
  - **Eco-Terrorist Leader**
    - 2x85% => 1x100% for one piece from each armor set
  - **Microtench**
    - 2x40% => 100%/40%
    - Ink, Poison moved to RDT @ 5%
  - **Octotench**
    - 2x40% => 100%/40%
    - Ink/Sac/Poison copied to RDT @ 5%
  - **Scorchellus**
    - 2x40% => 100%/40%
    - Map -> RDT @ 5%

### Hutlar
  - **Byysk Warrior**
    - 100% to drop one of: T3 elec, T3 fiberp, byysk meat, byysk tail
    - 50% to drop one of: Byysk gear
    - 10% (rare) to drop one of: Byysk gear
    - 5% (rare) to drop one of: map, citrine
    - 5%RDT Map:Citrine ratio to 4:1 from 8:1 
  - **Qion Slug**
    - 2x40% => 100%/40%
  - **Qion Tiger**
    - 2x40% => 100%/40%
    - tiger paw moved to RDT @ 5% (was 2/45)

### Tatooine
  - **Womprat**
    - 2x40% => 100%/40%
  - **Tusken Raider**
    - 2x40% => 100%/40%
    - Bread Flour + Rubies moved to RDT, weighted 2:3:2 Flour/Ruby/Maps
  - **Sandswimmer**
    - 2x40% => 100%/40%
  - **Sand Beetle**
    - 2x40% => 100%/40%
  - **Tusken Elite**
    - 2x40% => 100%/40%
    - Bread Flour, Rubies, and Hyphae Wood to RDT, weighted 2:4:2:1 Flour/Ruby/Wood/Maps
  - **Tusken Crate**
    - Cultured Butter + Dried Bonito marked as rare, affected by TH
  - **Sand Demon**
    - 2x40% => 100%/40%
    - Rubies moved to RDT @ 5%
  - **Sand Worm**
    - 2x60% => 100%/40%
    - Removed rare flag from sand worm teeth, rubies removed from drop table
    - Emeralds moved to RDT @ 5%
    - Emerald chance is 1/20 : ~1/6 : 1/4 : ~1/3 (TH levels 0 - 3)

### CODE REVIEW

File: Module/utc/sandswimmer.utc.json [hunk 1]
```diff
@@ -472,11 +472,11 @@
         "__struct_id": 4,
         "Spell": {
           "type": "word",
-          "value": 90
+          "value": 799
```
Comment (zunath) [inline Module/utc/sandswimmer.utc.json:475]: What was the spell change here and why was it made as part of the lot adjustments?

Comment (JohnWildkins) [reply Module/utc/sandswimmer.utc.json:475]: No idea, apparently the Toolset decided that Invisibility needed to be Vampire Invisibility

File: Module/utc/korr_hssiss.utc.json [hunk 2]
```diff
@@ -63,7 +63,7 @@
   "Description": {
     "type": "cexolocstring",
     "value": {
-      "0": ""
+      "0": "K'lor'slugs were dangerous vermiforms that possessed a serpentine body propelled by many dangerously-edged legs. Their mouthes are gaping circular maws filled with concentric rings of teeth. "
```
Comment (zunath) [inline Module/utc/korr_hssiss.utc.json:66]: Very minor typo here. "mouthes" -> "mouths"

File: SWLOR.Game.Server/Service/CraftService/RecipeType.cs [hunk 3]
```diff
@@ -716,7 +716,7 @@ public enum RecipeType
         WildBloodBroth = 2078,
         GrandioseSandwich = 2079,
         WildStew = 2080,
-        HssissSurprise = 2081,
+        KlorslugSurprise = 2081,
```
Comment (zunath) [inline SWLOR.Game.Server/Service/CraftService/RecipeType.cs:719]: This change is going to break. Database entities track recipes created by enumeration name and removing HssissSurprise is going to cause exceptions to be raised upon serialization/deserialization.

Simplest thing is to change this back. Less simple is to create a new recipe. Most complex is to do a data migration.

File: SWLOR.Game.Server/Service/QuestService/NPCGroupType.cs [hunk 4]
```diff
@@ -66,18 +66,18 @@ public enum NPCGroupType
         Hutlar_QionSlugs = 29,
         [NPCGroup("Qion Tigers")]
         Hutlar_QionTigers = 30,
-        [NPCGroup("Tuk'ata")]
-        Korriban_Tukata = 31,
-        [NPCGroup("Hssiss")]
-        Korriban_Hssiss = 32,
+        [NPCGroup("Pelko Bug Swarm")]
+        Korriban_PelkoSwarm = 31,
```
Comment (zunath) [inline SWLOR.Game.Server/Service/QuestService/NPCGroupType.cs:70]: This change will break in-progress quests. Again, we track data on entities by enumeration name. Simplest approach is to change these back to their original names. Most complex is to perform a data migration.

**Review by zunath — CHANGES_REQUESTED** [review]
Still need to address the data migration issues which will arise with the enumeration changes.

**Review by zunath — APPROVED** [review]
